### DISCUSSION TITLE
Auth mech for AWS Secret CSI Driver

### DISCUSSION THREAD

**Discussion by andrewgkew** [aws/secrets-store-csi-driver-provider-aws#245] [Q&A]
According to the docs and the more I play with this CSI driver it seems I can only authenticate with a role ie. IRSA but what if I want to use this on a local dev cluster like Kind?

What if I want to authenticate using an IAM user i.e. access and secret key? Can I not do that?

**Comment by andrewgkew** [comment]
Looks like according to this file that IRSA is the only way right now?

https://github.com/aws/secrets-store-csi-driver-provider-aws/blob/main/auth/auth.go

**Comment by joebowbeer** [comment]
Try mounting `~/.aws` to `/root/.aws` in the secrets-store-csi-driver pods.

Then ASCP may be able to assume a chained role (IRSA annotation) in a local cluster using your local `~/.aws/credentials`.

Related discussion: https://github.com/external-secrets/external-secrets/discussions/2567
